core/src/credential.rs

```diff
@@ -105,19 +202,35 @@ impl Credential {
     }
 }
 
-#[derive(Debug, serde::Serialize, serde::Deserialize, Clone)]
-pub struct VerifiablePresentation {
-    #[serde(flatten)]
-    pub presentation: Presentation,
-    pub proof: crate::proof::DataIntegrityProof,
+impl FromStr for Presentation {
+    type Err = super::error::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let p = Presentation::from_json_str(s)?;
+        Ok(p)
```

Comment:
same here, in general:

```
let a = some_func()?;
Ok(a)
```

can always be transformed into:
```
some_func()
```

(barring weirdness with `?` doing error type conversions)